src/solvers/dg.jl

```diff
@@ -261,30 +261,90 @@ Base.summary(io::IO, dg::DG) = print(io, "DG(" * summary(dg.basis) * ")")
 
 @inline Base.real(dg::DG) = real(dg.basis)
 
-@inline ndofs(mesh::TreeMesh, dg::DG, cache) = nelements(cache.elements) * nnodes(dg)^ndims(mesh)
-
 
 function get_element_variables!(element_variables, u, mesh, equations, dg::DG, cache)
   get_element_variables!(element_variables, u, mesh, equations, dg.volume_integral, dg, cache)
 end
 
 
+const MeshesDGSEM = Union{TreeMesh, StructuredMesh, UnstructuredMesh2D, P4estMesh}
+
+@inline ndofs(mesh::MeshesDGSEM, dg::DG, cache) = nelements(cache.elements) * nnodes(dg)^ndims(mesh)
```

Comment:
We could create a common interface for DG methods if `nnodes(dg)^ndims(mesh)` was replaced by a more general `nnodes` which returns the number of nodes on an element. This definitely wouldn't be something for this PR (and it's not really an issue due to dispatch on `MeshesDGSEM`), but I thought it would be good to discuss if a more unified interface would be useful.

Reply:
Could you please open an issue for that?

Reply:
See https://github.com/trixi-framework/Trixi.jl/issues/711

Reply:
Thanks!